lib/_stream_writable.js

```diff
@@ -449,8 +487,12 @@ Writable.prototype.end = function(chunk, encoding, cb) {
     encoding = null;
   }
 
+  state.flushing = true;
+
   if (chunk !== null && chunk !== undefined)
     this.write(chunk, encoding);
+  else if (state.stringDecoder && state.stringDecoder.lastNeed > 0)
+    this.write(Buffer.alloc(0));
```

Comment:
These changes to `.end()` also put the source size over the default inlineability limit. One suggestion that would keep it just under the limit is to s/encoding/enc/ and use a shared/static `Buffer.alloc(0)` (declare it in the "global" scope with a small enough name and just reference it here). I'm not sure what, if any issues may arise from a shared/static empty Buffer though...